url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/dataStructure/AdjList.java

```diff
@@ -0,0 +1,352 @@
+    			temp = nodes;
+        		while(nodes != null) {
+        			if (nodes.getVid() == v) {
+        				temp.setProx(nodes.getProx());
+        				nodes = null;
+        				break;
+        			}
+        			temp = nodes;
+        			nodes = nodes.getProx();
+        		}    			
+    		}
+    	}
+    }
+    
+    public void insertEdge (int v1, int v2, double weight) {
+    	ALNode node = new ALNode (v2,weight);
+
+    	node.setProx(adjl[v1]);
+    	adjl[v1] = node;
+        }    
+    
+    public void checkInsertEdge (int v1, int v2, double weight) {
+    	ALNode temp = adjl[v1];
+    	while (temp != null) {
+    		if (temp.getVid() == v2)
+    			return;
+    		temp = temp.getProx();
+    	}
+    	insertEdge(v1, v2, weight);
+    }    
+    
+    public void printTree(int root) {
+    	ArrayList<Integer> levels = new ArrayList<Integer>();
+    	Stack<Integer> stack = new Stack<Integer>();
+    	levels.add(root);
+    	System.out.println("imprimindo árvore:");
+    	while (!levels.isEmpty()) {
+    		int index = levels.remove(0);
+    		ALNode node = adjl[index];
+    		if (node == null)
+    			continue;
+    		System.out.print(index+": [");
+    		while (node != null) {
+    			levels.add(node.getVid());
+    			System.out.print(node.getVid()+", ");
+    			node = node.getProx();
+    		}
+    		System.out.println("]");
+    	}
+    }
+
+    public void printAdjList() {
+	int i;
+	ALNode node;
+
+	System.out.println ("printing Adjacent List:");
+	for (i=0; i < nv; i++) {
+		if (adjl[i] == null)
+			continue;
+	    System.out.print ("Vertice: " + i + ": ");
+	    node = adjl[i];
+	    while (node != null) {
+	    	System.out.print(node.getVid()+"/c: "+node.getWeight()+", ");
+		node = node.getProx();
+	    }
+	    System.out.println();
+	}
+    }
+
+    public void setWeight(Weight w) {
+	ALNode node;
+
+	for (int i=0; i < nv; i++) {
+	    node = adjl[i];
+	    while (node != null) {
+		node.setWeight(w.getWeight(i, node.getVid(), this));
+		node = node.getProx();
+	    }
+	}
+    }
+
+    public  void checkCopyList (AdjList nadjl, int index) {
+    	ALNode p;
+
+    	for (ALNode q = nadjl.getList(index); q != null; q = q.getProx()) {
+    		ALNode temp = adjl[index];
+    		boolean flag = false;
+    		while (temp != null) {
+    			if (temp.getVid() == q.getVid()) {
+    				flag = true;
+    				break;
+    			}
+    			temp = temp.getProx();
+    		}
+    		if (flag)
+    			continue;
+    	    p = new ALNode(q.getVid(), q.getWeight());
+    	    p.setProx(adjl[index]);
+    	    adjl[index] = p;
+    	}
+    }
+
+    public  void copyList (AdjList nadjl, int index) {
+	ALNode p;
+
+	for (ALNode q = nadjl.getList(index); q != null; q = q.getProx()) {
+	    p = new ALNode(q.getVid(), q.getWeight());
+	    p.setProx(adjl[index]);
+	    adjl[index] = p;
+	}
+    }
+
+
+    public void readFromTopologyFile (String fname, PositionTable pt) {
+	String s1, s2;
+
+	try {
+	    BufferedReader in = new BufferedReader(new FileReader(fname));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 236, 240, 241, 263. The resource is closed at line 265. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).